aws/s3.go

```diff
@@ -20,37 +21,49 @@ type S3Config struct {
 	SecretAccessKey string `json:"secret_access_key"`
 	Bucket          string `json:"bucket"`
 	Path            string `json:"path"`
+	ForcePathStyle  bool   `json:"force_path_style"`
 }
 
 // S3Client is a client for uploading data to S3.
 type S3Client struct {
-	endpoint  string
-	region    string
-	accessKey string
-	secretKey string
-	bucket    string
-	key       string
+	endpoint       string
+	region         string
+	accessKey      string
+	secretKey      string
+	bucket         string
+	key            string
+	forcePathStyle bool
```

Comment:
From reading this: https://docs.aws.amazon.com/sdk-for-go/api/aws/#Config

whether S3ForcePathStyle is set or not, uploading will work, correct? I'm reading the AWS SDK docs, but it's not clear what the implications are of setting this variable. This is distinct from this PR, does it matter if this variable is set or not?

Reply:
Amazon S3 works fine when it's set to true, yeah, but Amazon has deprecated this mode and is phasing it out, so for Anazon-native S3 I would want to enable it even if it does currently work.

https://aws.amazon.com/blogs/aws/amazon-s3-path-deprecation-plan-the-rest-of-the-story/